### PR TITLE
feat: CRN-853 Add PUT endpoint for RO

### PR DESCRIPTION
**Pull request by peterstarling**

### CODE REVIEW

**Review by niezgoda — APPROVED** [review]

File: packages/squidex/src/utils.ts [hunk 1]
```diff
@@ -29,12 +29,17 @@ export type RequiredAndNonNullable<T> = Required<
...
-}): SquidexEntity =>
-  Object.entries(object).reduce((acc, [key, value]) => {
-    acc[key] = { iv: value };
-    return acc;
-  }, {} as SquidexEntity);
+export const parseToSquidex = <T>(object: T): SquidexEntityObject<T> =>
+  Object.entries(object).reduce(
+    (acc, [key, value]) => ({
+      ...acc,
+      [key]: typeof value === 'undefined' ? undefined : { iv: value },
```
[the remark this answers is not in the document]
Comment (peterstarling) [reply packages/squidex/src/utils.ts]: I have added some tests to make sure we know what the desired behaviour is, I have also compared against what squidex input should look like. Basically undefined should be removed from the var whereas nulls should be given as iv: null which is what it does now

File: apps/crn-server/src/validation/research-output.validation.ts [hunk 2]
```diff
@@ -139,6 +140,9 @@ const researchOutputPostRequestValidationSchema: JSONSchemaType<ResearchOutputPo
       'sharingStatus',
       'addedDate',
       'teams',
+      'methods',
```
Comment (niezgoda) [inline apps/crn-server/src/validation/research-output.validation.ts:143]: there is two tasks on the fe part - so when released prior to them - adding RO is going to stop working

File: apps/crn-server/src/routes/research-outputs.route.ts [hunk 3]
```diff
@@ -62,42 +59,35 @@ export const researchOutputRouteFactory = (
...
   });
 
+  researchOutputRoutes.put(
+    '/research-outputs/:researchOutputId',
+    async (req, res) => {
+      const { body, params, loggedInUser } = req;
+      const { researchOutputId } = validateResearchOutputParameters(params);
+      const updateRequest = validateResearchOutputPutRequestParameters(body);
+      validateResearchOutputPostRequestParametersIdentifiers(body);
+
```
Comment (niezgoda) [inline apps/crn-server/src/routes/research-outputs.route.ts:78]: you are not checking the  hasCreateUpdateResearchOutputPermissions

File: apps/crn-server/src/controllers/research-outputs.ts [hunk 4]
```diff
@@ -147,79 +152,82 @@ export default class ResearchOutputs implements ResearchOutputController {
...
-      subtype,
-    });
+      researchOutputId = result.id;
 
-    await Promise.all(
-      teams.map((teamId) =>
-        this.associateResearchOutputToTeam(teamId, researchOutputId),
-      ),
-    );
+      await Promise.all(
```
Comment (markcsims) [inline apps/crn-server/src/controllers/research-outputs.ts:225]: I know this isn't from this PR, but maybe one to discuss. 

We're using a Promise.all, and which will reject upon first failure.

**Review by niezgoda — APPROVED** [review]

**Pull request merged**